api/oc_events.h

```diff
@@ -52,6 +52,15 @@ typedef enum {
 
 extern oc_process_event_t oc_events[];
 
+/**
+ * @brief convert oc_events_t value to oc_process_event_t value
+ *
+ * @param event value to convert
+ * @return oc_process_event_t on success
+ * @return -1 on error
+ */
+int oc_event_to_oc_process_event(oc_events_t event);
```

Comment:
events are used for internal flow so I don't want to expose them because you need to know how can use them.

Reply:
why it returns `int` type?

Reply:
So I should move this function somewhere in include? I'm using this in my binary so I need a public function.

It returns because `oc_process_event_t` is `unsigned char` and I need `-1` for error (in case event of given type has not been allocated). Items in `oc_events[]` are initialized by this function:

https://github.com/iotivity/iotivity-lite/blob/master/util/oc_process.c#L90-L94

In case that we know that lastevent will never reach 255 then I can use `oc_process_event_t` as return type and just do `(oc_process_event_t)-1` for error return.